Tests/FluentAssertions.Specs/Specialized/TaskAssertionSpecs.cs

```diff
@@ -180,6 +184,274 @@ public async Task When_task_completes_late_it_should_succeed()
         }
     }
 
+    public class ThrowAsync
+    {
+        [Fact]
+        public async Task When_subject_is_null_it_should_throw()
+        {
+            // Arrange
+            Func<Task> action = null;
+
+            // Act
+            Func<Task> testAction = () => action.Should().ThrowAsync<InvalidOperationException>(
+                "because we want to test the failure {0}", "message");
+
+            // Assert
+            await testAction.Should().ThrowAsync<XunitException>()
+                .WithMessage("*because we want to test the failure message*found <null>*");
+        }
+
+        [Fact]
+        public async Task When_subject_is_null_in_assertion_scope_it_should_throw()
+        {
+            // Arrange
+            Func<Task> action = null;
+
+            // Act
+            Func<Task> testAction = async () =>
+            {
+                using var _ = new AssertionScope();
+                await action.Should().ThrowAsync<InvalidOperationException>(
+                    "because we want to test the failure {0}", "message");
+            };
+
+            // Assert
+            await testAction.Should().ThrowAsync<XunitException>()
+                .WithMessage("*because we want to test the failure message*found <null>*");
+        }
+
+        [Fact]
+        public async Task When_task_throws_it_should_succeed()
+        {
+            // Arrange
+            var timer = new FakeClock();
+            var taskFactory = new TaskCompletionSource<bool>();
+
+            // Act
+            Func<Task> action = () =>
+            {
+                return taskFactory
+                    .Awaiting(t => (Task)t.Task)
+                    .Should(timer).ThrowAsync<InvalidOperationException>();
+            };
+            taskFactory.SetException(new InvalidOperationException("foo"));
+
+            // Assert
+            await action.Should().CompleteWithinAsync(1.Seconds());
+        }
```

Comment:
Disclaimer, I often struggle with mentally keeping track of everything our timing related tests...

But shouldn't we use `timer` for anything? 🤔 
Otherwise, would the snippet below suffice?

```cs
[Fact]
public async Task When_task_throws_it_should_succeed()
{
    // Arrange
    var taskFactory = new TaskCompletionSource<bool>();

    // Act
    Func<Task> action = () =>
    {
        return taskFactory
            .Awaiting(t => (Task)t.Task)
            .Should().ThrowAsync<InvalidOperationException>();
    };
    taskFactory.SetException(new InvalidOperationException("foo"));

    // Assert
    await action.Should().NotThrowAsync();
}
```

Reply:
`CompleteWithinAsync` and `NotThrowAsync` has the same meaning but `CompleteWithinAsync` defines a timeout for the test.
But you are right, the timer can be removed here.